packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue

```diff
@@ -165,26 +181,80 @@ const openMatrixModal = (datum: ModelConfigTableData) => {
 
 const rowClass = (rowData) => (rowData.type === ParamType.MATRIX ? '' : 'no-expander');
 
+const updateTimeseries = (id: string, value: string) => {
+	if (!validateTimeSeries(value)) return;
+	const clonedConfig = cloneDeep(props.modelConfiguration);
+	clonedConfig.configuration.metadata.timeseries[id] = value;
+	emit('update-configuration', clonedConfig);
+};
+
+const validateTimeSeries = (values: string) => {
+	let isValid = true;
+	if (typeof values !== 'string') {
+		isValid = false;
+		errorMessage.value = 'Incorrect Format (e.g., 0:500, 10:550, 25:700 etc)';
+		return isValid;
+	}
+	const timeValuePairs = values.split(',');
+
+	timeValuePairs.forEach((pair) => {
+		const colonCount = pair.split(':').length - 1;
+
+		// If a pair does not contain exactly one colon, return false
+		if (colonCount !== 1) {
+			isValid = false;
+			return;
+		}
+
+		const [time, value] = pair.split(':').map((part) => part.trim());
+
+		if (!/^\d+$/.test(time) || !/^\d+$/.test(value)) {
+			isValid = false;
+		}
+	});
+
+	errorMessage.value = '';
+	if (!isValid) {
+		errorMessage.value = 'Incorrect Format (e.g., 0:500, 10:550, 25:700 etc)';
+	}
+	return isValid;
+};
```

Comment:
You could simplify a lot this function now that you defined it.

```suggestion
const validateTimeSeries = (values: string): boolean => {
    const message = 'Incorrect Format (e.g., 0:500, 10:550, 25:700 etc)';
    
    if (typeof values !== 'string') {
        errorMessage.value = message;
        return false;
    }

    const isPairValid = (pair: string): boolean => /^\d+:\d+$/.test(pair.trim());
    const isValid = values.split(',').every(isPairValid);

    errorMessage.value = isValid ? '' : message;
    return isValid;
};
```

Reply:
changed